### ISSUE TITLE
using deep-assign package

### ISSUE THREAD

**Issue by ndamjan** [feathersjs/feathers-vuex#15]
I stumbled upon some strange exception where a nested field from a JSON object was missing... Can't reproduce it currently in JSFiddle.

To me it seems some like cloning issue so I checked where it happens, and the responsibility is on deep-assign package.

According to https://github.com/sindresorhus/deep-assign:
> [DON'T USE THIS MODULE PLEASE]

Replacing it with object-assign doesn't work.  I will see some alternatives and investigate further...

**Comment by marshallswain** [comment]
LOL.  That kinda of message ought to be in the readme.  Thanks for finding that.

**Comment by ndamjan** [comment]
I found one more issue related to deep-assign - it doesn't make a deep clone of an object. This manifested itself as having the same refererence for an object property within both `copy` and `current`. So when I made changes to copy, they were being made to current also although the copy was not commited.

My workaround for this is to change a function in **service-module/mutations.js** to use clone function from [ramda](http://ramdajs.com/docs/#clone) instead of deepAssign.

```javascript
    setCurrent (state, item) {
      let id = isObject(item) ? item[idField] : item
      state.currentId = id
      // state.copy = deepAssign({}, item)
      state.copy = R.clone(item)
    },
```

**Comment by ndamjan** [comment]
Found also `commitCopy` and `rejectCopy` to be problematic because of `deepAssign`. I replace it the same way. Should we go for PR with this?

**Comment by marshallswain** [comment]
@ndamjan  Thanks for discovering all of this.  Since the cloning is the only part of Ramda that we would be using, I'd prefer to go with a standalone solution like https://www.npmjs.com/package/lodash.clonedeep to keep the package size smaller.

**Comment by daffl** [comment]
Closed via #35 and released as v0.8.0
